scripts/functions.js

```diff
@@ -760,12 +758,18 @@ function globalUnlock(key) {
     sys.storage.remove(key);
 }
 
-function getNonce(address) {
-    var lastNonce = sys.storage.get("ethereum-endpoint-"+address+'-nonce');
-    if (lastNonce) {
-        var newNonce = parseInt(lastNonce) + 1;
-        return '0x'+newNonce.toString(16);
+function getNonce(address, nonceCalculationMethod) {
+    var nonce = sys.storage.get("ethereum-endpoint-"+address+'-nonce');
+    if (nonce && nonceCalculationMethod == 'calculationInEndpoint') {
+        return nonce;
     } else {
         return endpoint.eth.transactionCount(address, 'pending');
```

Comment:
I would also clear the nonce in the storage if it is not null. This is because if the user is sometimes calling with `calculationInEndpoint` and sometimes without it, the nonce we have in the storage could get outdated and we should force to use `transactionCount()` and store it in the storage again.